astropy/cosmology/flrw/base.py

```diff
@@ -1450,6 +1450,71 @@ def equivalent_nonflat(self: _FlatFLRWMixinT) -> _FLRWT:
 
         return inst
 
+    def clone(self, *, meta: Optional[Mapping]=None, to_nonflat: bool=False, **kwargs: Any):
+        """Returns a copy of this object with updated parameters, as specified.
+
+        This cannot be used to change the type of the cosmology, except for
+        changing to the non-flat version of this cosmology.
+
+        Parameters
+        ----------
+        meta : mapping or None (optional, keyword-only)
+            Metadata that will update the current metadata.
+        to_nonflat : bool, optional keyword-only
+            whether to change to the non-flat version of this cosmology.
+        **kwargs
+            Cosmology parameter (and name) modifications.
+            If any parameter is changed and a new name is not given, the name
+            will be set to "[old name] (modified)".
+
+        Returns
+        -------
+        newcosmo : `~astropy.cosmology.Cosmology` subclass instance
+            A new instance of this class with updated parameters as specified.
+            If no modifications are requested, then a reference to this object
+            is returned instead of copy.
+
+        Examples
+        --------
+        To make a copy of the ``Planck13`` cosmology with a different matter
+        density (``Om0``), and a new name:
+
+            >>> from astropy.cosmology import Planck13
+            >>> Planck13.clone(name="Modified Planck 2013", Om0=0.35)
+            FlatLambdaCDM(name="Modified Planck 2013", H0=67.77 km / (Mpc s),
+              Om0=0.35, ...
+
+        If no name is specified, the new name will note the modification.
+
+            >>> Planck13.clone(Om0=0.35).name
+            'Planck13 (modified)'
+
+        The keyword 'to_nonflat' can be used to clone on the non-flat
+        equivalent cosmology.
+
+            >>> Planck13.clone(to_nonflat=True)
+            LambdaCDM(name="Planck13", ...
+
+            >>> Planck13.clone(H0=70, to_nonflat=True)
+            LambdaCDM(name="Planck13 (modified)", H0=70.0 km / (Mpc s), ...
+
+        This allows ``Ode0`` to be modified.
+
+            >>> Planck13.clone(to_nonflat=True, Ode0=1)
+            LambdaCDM(name="Planck13 (modified)", H0=67.77 km / (Mpc s),
+                      Om0=0.30712, Ode0=1.0, ...
+
+        Since ``Ode0`` cannot be given to any flat cosmology, if it is a kwarg
+        then the cosmology is assumed to not be flat so ``to_nonflat`` is not
+        required.
+
+            >>> Planck13.clone(Ode0=1)
+            LambdaCDM(name="Planck13 (modified)", H0=67.77 km / (Mpc s),
+                      Om0=0.30712, Ode0=1.0, ...
+        """
+        to_nonflat = to_nonflat or ("Ode0" in kwargs)
```

Comment:
This feels wrong: if `to_nonflat=False`, shouldn't one get an error when passing in `Ode0`?

Reply:
A good point. I'm changing it so that `to_nonflat` is default None, in which case it looks for `"Ode"` in the kwargs. If it's a boolean than that is its value and if False and Ode0 is given it will error.